src/main/java/com/michelin/cio/hudson/plugins/maskpasswords/MaskPasswordsBuildWrapper.java

```diff
@@ -108,7 +110,31 @@ public OutputStream decorateLogger(AbstractBuild build, OutputStream logger) {
             }
         }
 
-        return new MaskPasswordsOutputStream(logger, allPasswords);
+        return new FilterImpl(allPasswords);
+    }
+
+    private static final class FilterImpl extends ConsoleLogFilter implements Serializable {

```

Comment:
Maybe we should promote `@PersistedIn` annotation so that we can use it elsewhere. I find the designation like that useful, as persistence situation affects migration/evolution story of the class.